middleman-core/features/cli/preview_server.feature

```diff
@@ -0,0 +1,376 @@
+    """
+
+  Scenario: Start the server with bind address 127.0.0.1
+    When I run `middleman server --bind-address 127.0.0.1` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    Inspect your site configuration at "http://127.0.0.1:4567/__middleman"
+    """
+
+  Scenario: Start the server with bind address 127.0.0.1 configured via config.rb
+    Given a file named "config.rb" with:
+    """
+    set :bind_address, '127.0.0.1'
+    """
+    When I run `middleman server` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://127.0.0.1:4567"
+    """
+    And the output should contain:
+    """
+    Inspect your site configuration at "http://127.0.0.1:4567/__middleman"
+    """
+
+  Scenario: Start the server with bind address 127.0.0.5
+    When I run `middleman server --bind-address 127.0.0.5` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://127.0.0.5:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://127.0.0.5:4567"
+    """
+    And the output should contain:
+    """
+    Inspect your site configuration at "http://127.0.0.5:4567/__middleman"
+    """
+
+  Scenario: Start the server with bind address ::1
+    Given a file named ".hosts" with:
+    """
+    # <ip-address> <hostname.domain.org> <hostname>
+    ::1 localhost.localdomain localhost
+    """
+    When I run `middleman server --bind-address ::1` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://::1:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://::1:4567"
+    """
+    And the output should contain:
+    """
+    Inspect your site configuration at "http://::1:4567/__middleman"
+    """
+
+  Scenario: Start the server with bind address 0.0.0.0
+    When I run `middleman server --bind-address 0.0.0.0` interactively
+    And I stop middleman if the output contains:
+    """
+    Inspect your site configuration
+    """
+    Then the output should contain:
+    """
+    The Middleman is listening on "http://0.0.0.0:4567"
+    """
+    And the output should contain:
+    """
+    View your site at "http://
```

Comment:
`0.0.0.0` is not a valid IP, it's only routed to `localhost` on Linux - this URL doesn't work on Windows.